CONTRIBUTING.md

```diff
@@ -3,62 +3,75 @@
 - [cfn-python-lint](https://github.com/aws-cloudformation/cfn-python-lint)
-- [cnf-nag](https://github.com/stelligent/cfn_nag)
-- [EditorConfig](https://editorconfig.org/)
+- [RuboCop](https://rubocop.org)
 
-### Code Style
+You should set up realtime checking with these tools in your IDE, such as type checking with TypeScript and linting with ESLint. You should also enable Prettier to format in realtime or on save. The test suite will fail if there are rules violations, so it's much easier to be validation code changes as you go.
 
-This repository contains an [EditorConfig](https://editorconfig.org/) file that dictates some basic code styles. The repository may also contain editor- or IDE-specific settings files, which must follow the styles defined in the EditorConfig file. Any other linters or formatting tools should also follow those styles. You should ensure that your editor is setup to use the EditorConfig styles, regareless of how those settings are installed into program.
+While the project utilizes TypeScript for type checking and other other JavaScript syntax checking, Node.js-relate code should be written in vanilla JavaScript, and not TypeScript.
 
-For YAML, JavaScript, and JSON files your editor should be configured to _format on save_, and should use [Prettier](https://prettier.io/) as the formatter, following the settings described in `.prettierrc`. Ensure that the state machine files (`.asl.json`) are also being formatted.
+### Commands & Toolchain
 
-[ESLint](https://eslint.org/) is used for JavaScript files. The rules are primarily inherited from the Airbnb rules and made to work with the Prettier settings, with some minor project-specific modifications. If possible, it can be helpful to have your editor lint JavaScript files on the fly.
+The project includes a `Makefile` to centralize various commands used in development and operation. When new commands are required, be sure to add them to the `Makefile`, even if it calls out to something else, like `npm run-script`.
 
-CloudFormation templates should be run through [cfn-python-lint](https://github.com/aws-cloudformation/cfn-python-lint) and be free of linting errors. Your editor should check your files in realtime if possible; there are plugins available for many popular editors.
+Besides language runtimes, package managers, and command line programs, tooling should be written to assume that libraries are installed within the package, not globally available on the system. I.e, do not asusme the `eslint` or `prettier` are installed globally.
 
-### AWS
+## Getting Started
 
-There are a number of shell scripts in this repo. They assume you have a profile in `~/.aws/config` called `prx-legacy`, which is configured to access the `prx-legacy` AWS account.
+After cloning the Git repository for Porter, run `make bootstrap` to install all necessary development dependencies. This requires already having Node.js, Ruby, and Python 3 installed on your system.
```

Comment:
Might be a [bit more typical](https://www.gnu.org/prep/standards/html_node/Standard-Targets.html) to use an `install` target.

But since this isn't really "compiling" anything, maybe that's a bad metaphor.

Reply:
I use the `INSTALL` doc for information about deploying the app, rather than setting it up for dev, so it would be overloading the term a bit to use it both ways. Good point, though.